### PR TITLE
sql: support `SHOW BACKUP METADATA FROM <storage:str>`

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42913

Problem Summary:
This PR supports the `SHOW BACKUP METADATA FROM` SQL syntax.

### What is changed and how it works?
This PR added a SQL for fetching the backup metadata from an external storage.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Added the `SHOW BACKUP METADATA FROM <storage:str>` syntax support.
```

### CODE REVIEW

File: executor/brie.go [hunk 1]
```diff
@@ -353,16 +360,68 @@ type BRIEExec struct {
...
+func (e *BRIEExec) fillByShowMetadata(s *ast.BRIEStmt) {
+	if s.Kind != ast.BRIEKindShowBackupMeta {
+		panic(fmt.Sprintf("precondition failed: `fillByShowMetadata` should always called by a ast.BRIEKindShowBackupMeta, but it is %s.", s.Kind))
+	}
+
+	store := s.Storage
+	e.showConfig = &show.Config{
+		Storage: store,
+		Cipher: backuppb.CipherInfo{
+			CipherType: encryptionpb.EncryptionMethod_PLAINTEXT,
```
Comment (BornChanger) [inline executor/brie.go:376]: So, only plain text encryption support now?

**Review by BornChanger — APPROVED** [review]
lgtm

File: br/pkg/metautil/metafile.go [hunk 2]
```diff
@@ -274,31 +274,57 @@ func (reader *MetaReader) ReadDDLs(ctx context.Context) ([]byte, error) {
...
-		tableID := tablecodec.DecodeTableID(file.GetStartKey())
-		if tableID == 0 {
-			log.Panic("tableID must not equal to 0", logutil.File(file))
+	var fileMap map[int64][]*backuppb.File
+	if !cfg.skipFiles {
+		fileMap = make(map[int64][]*backuppb.File)
+		outputFn := func(file *backuppb.File) {
+			tableID := tablecodec.DecodeTableID(file.GetStartKey())
+			if tableID == 0 {
+				log.Panic("tableID must not equal to 0", logutil.File(file))
```
Comment (Leavrth) [inline br/pkg/metautil/metafile.go:320]: make `TiDB` panic when use brie SQL :dizzy_face:

File: br/pkg/task/show/cmd.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	StartKey     logutil.HexBytes `json:"start-key"`
+	EndKey       logutil.HexBytes `json:"end-key"`
+}
+
+type Table struct {
+	DBName         string `json:"db-name"`
+	TableName      string
+	KVCount        uint
+	KVSize         uint
+	TiFlashReplica uint
```
Comment (Leavrth) [inline br/pkg/task/show/cmd.go:49]: miss `json: ...`(otherwise it would use default json name) or `json:"-"`

**Review by Leavrth — APPROVED** [review]

File: br/pkg/metautil/metafile.go (hunk 2 above)
Comment (YuJuncen) [reply br/pkg/metautil/metafile.go:320]: Don't worry, TIDB will catch this panic...

File: br/pkg/task/show/cmd.go (hunk 3 above)
Comment (YuJuncen) [reply br/pkg/task/show/cmd.go:56]: 🤣

**Pull request merged**